jlm/hls/backend/rvsdg2rhls/add-sinks.cpp

```diff
@@ -47,7 +48,8 @@ add_sinks(llvm::RvsdgModule & rm)
 {
   auto & graph = rm.Rvsdg();
   auto root = &graph.GetRootRegion();
-  add_sinks(root);
+  auto lambda = dynamic_cast<rvsdg::LambdaNode *>(root->Nodes().begin().ptr());
```

Comment:
Maybe also add an assert that the root region really only contains a single node.

Reply:
Added